borg/crypto.pyx

```diff
@@ -136,3 +155,16 @@ cdef class AES:
             return out[:ptl]
         finally:
             free(out)
+            PyBuffer_Release(&data_buf)
+
+
+def hmac_sha256(key,  data):
+    cdef Py_buffer data_buf = ro_buffer(data)
+    md = bytes(32)
+    try:
+        if not HMAC(EVP_sha256(), <const unsigned char*> key, len(key),
```

Comment:
I haven't quite understood what OpenSSL does there (and the code is very opaque). My understanding from the docs and their examples is, that EVP_sha256() or EVP_aes_whatever_cbc() are just functions that return a pointer to some static structure in the OpenSSL library that just contains further pointers to functions actually implementing these algorithms. I don't think these should be free()d.

Interestingly enough none of these functions are actually defined in the OpenSSL source tree, not even after compilation, so they are probably generated and put somewhere outside the tree.